### DISCUSSION TITLE
Custom Month and Year dropdown using react-select

### DISCUSSION THREAD

**Discussion by brendonco** [gpbl/react-day-picker#1512] [Support] [answered]
Is there a way to custom the dropdown to use react-select?

https://react-day-picker.js.org/basics/navigation#using-a-drop-down-to-change-the-month

**Comment by gpbl** [comment, marked as answer]
@brendonco never tried it with react-select, but you can try implementing your own dropdown component using it, and pass it to the `components` prop:

```tsx
import { DropdownProps } from 'react-day-picker';
function CustomDropDown(props: DropdownProps) {
  return <ReactSelect/>
}
export function CustomDayPicker() {
  return (
    <DayPicker
       components={{
         Dropdown: CustomDropDown
       }}
    />;
  );
}
```

https://react-day-picker.js.org/guides/custom-components
https://react-day-picker.js.org/api/interfaces/CustomComponents#dropdown

Let us know how far you go!

**Reply by brendonco** [reply]
Hi @gpbl, Thanks. It works. Unfortunately there are couple of minor functionality got affected by this change.

1. For Month/Year selections, which hooks can I use to bind to onChange event? I tried `useNavigation` e.g. `gotoMonth(7)`, not working for changing the month/year.
2. The arrow navigation are gone after custom dropdown is enabled. How do I bring it back? 

![image](https://user-images.githubusercontent.com/6521691/185524477-be99412d-9844-4431-960b-32d722b1272f.png)

**Reply by gpbl** [reply]
@brendonco I think you want to call the `useNavigation` hook. It returns an object with `goToMonth()`, which should be called when the select changes. 

https://react-day-picker.js.org/guides/custom-components#daypicker-hooks

Regarding the arrows - these are actually _replaced_ by the dropdown. In the original design, you can't choose both because there is not enough horizontal space.

**Reply by gpbl** [reply]
You could also implement a custom caption component, where you would use react-selects, or whatever you prefer in the caption. 

https://react-day-picker.js.org/guides/custom-components#custom-caption

**Reply by brendonco** [reply]
@gpbl I'm able to navigate Month and Year using useNavigation e.g. gotoMonth(7). But when I select a Day. e.g. January 31, 2015. onDayClick, the day is pointing to 'Jan 31 2022' instead of 'Jan 31 2015'. Any idea how to synced up with the calendar day picker?

```
<DayPicker
  onDayClick={onDayClick}

```

```
// Day - 'Jan 31 2022' instead of 'Jan 31 2015'
  const onDayClick = (day, { selected }) => {
```

**Reply by gpbl** [reply]
@brendonco I can't help without a code example. We recently released a version that should fix a problem with `goToMonth`: try to upgrade and see if your issue is still there!

**Reply by brendonco** [reply]
@gpbl With Custom Month and Year dropdown, if I select Year Dropdown, how do I get the current Month dropdown selected value and vice versa? I can't seem to find a document.

Currently doing querySelector to get the value. If you have a document on how to get the current selected Month or Year, please do share. :)

`document.querySelector('#MonthsDropdown')?.textContent;`

**Reply by gpbl** [reply]
@brendonco it's difficult to understand what you are doing, without some code examples. Would you mind to for this code sandbox and provide some code? https://codesandbox.io/s/react-day-picker-8-vl4ry

Ideally, you would call `goToMonth` when the selects change. This will set the `currentMonth` in the calendar that you can use to know which month or year could be selected:

[react-day-picker.js.org/guides/custom-components#custom-caption](https://react-day-picker.js.org/guides/custom-components#custom-caption)

**Reply by brendonco** [reply]
@gpbl After Upgrade to latest version, gotoMonth is working.  One more thing, does multiple day selection work with `captionLayout: dropdown` ?  Multiple day selection doesnt seem to work.

```
 <DayPicker
     captionLayout="dropdown"
     fromDate={new Date()}
     toDate={some date}
      mode="multiple"
      min={1}
      selected={days}
      onSelect={setDays}
      footer={footer}
    />
```

**Reply by gpbl** [reply]
@brendonco no reason it should not work, please reproduce your issue in https://codesandbox.io/s/react-day-picker-8-vl4ry and open a new discussion or bug if you can reproduce it, thanks

**Reply by brendonco** [reply]
> ```tsx
> components={{
>          Dropdown: CustomDropDown
>        }}
> ```

Here's the sample code on jumping to Year or Month using `document.querySelector('#MonthsDropdown')?.textContent;`  

https://replit.com/@brendonco/React-Day-Picker-Multi-Day-Selection#src/App.tsx

Do you have similar API instead of using querySelector to jump to Month/Year? In order to jump to a Year with Month not selected, you have to get the current month selected and trigger gotoMonth. Any suggestion?
